### PR TITLE
Draft: Feature/sidebar

### PR DESCRIPTION
**Pull request by adamdavies1**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Please also include any acceptance criteria if you have any.

- Added the ability to add a sidebar component to the left and/or right hand side of the page.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce
the testing if necessary.

Before PR's can be merged they will need to be tested by QA and approved where
applicable. To flag the change to QA assign **@XGovFormBuilder/qa** as one of the reviewers.

- [ ] Create a new form, click create component on a page. You should see a new Sidebar component. This will take html and can be aligned left or right. Save changes a preview this in the runner

# Checklist:

- [x] My changes do not introduce any new linting errors
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto main and there are no code conflicts
- [x] I have checked deployments are working in all environments
- [ ] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Comment by superafroman** [comment]
@adamdavies1 I'm interested in the user needs behind this feature. When would a side bar be used?

File: runner/src/server/plugins/engine/views/index.html [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
-  <div class="govuk-grid-row">
+<div class="govuk-grid-row"> 
+  {% set count = 0 %} 
+  {% for component in components %}
+    {%if component.type.toLowerCase() == "sidebar" %}
+     {% set count = count + 1 %}
+    {% endif %}    
+  {% endfor %}
+
+  {% set gridSize = "one-half" if count == "2" else "two-thirds" %}
```
Comment (adamdavies1) [inline runner/src/server/plugins/engine/views/index.html:17]: I don't really like what i have done here. It was the only way i was able to get the gridsize and alignment right.
When using dev tools to set the alignment in the css, i discovered that all sidebars would inherit the alignment regardless if they were unique or not.

**Comment by adamdavies1** [comment]
> @adamdavies1 I'm interested in the user needs behind this feature. When would a side bar be used?

Sorry for the delay @superafroman. Was given this task by @endamccormack (tagged incase more input needed). Basically it can be used for navigating to useful links related to the application. Heres an [example](https://www.gov.uk/sign-in-childcare-account)

**Comment by JamesBoadi** [comment]
@jenbutongit Can be closed

**Pull request closed without merging**